src/main/java/se/liu/ida/hefquin/engine/federation/catalog/FederationDescriptionReader.java

```diff
@@ -71,6 +71,15 @@ public FederationCatalog parseFedDescr( final Model fd ) {
 
 			// Check the type of interface
 			if ( ifaceType.equals(FD.SPARQLEndpointInterface) ) {
+
+				int numberOfEndpointAddresses = iface.listProperties(FD.endpointAddress).toList().size();
```

Comment:
Simply use `iface.listProperties(FD.endpointAddress)` only and then use the `StmtIterator` that it returns.

Reply:
I applied that and used the iterator instead of ".toList().size()".